tests/testthat/test-shinytest2-teal.R

```diff
@@ -0,0 +1,368 @@
+  expect_identical(
+    get_active_selection_value(app, "iris", "Species", module_id = "module_1"),
+    c("setosa", "versicolor", "virginica")
+  )
+  expect_identical(
+    get_active_selection_value(app, "mtcars", "cyl", module_id = "module_1"),
+    c("4", "6")
+  )
+  expect_null(get_active_selection_value(app, "mtcars", "drat", is_numeric = TRUE, module_id = "module_1"))
+  expect_null(get_active_selection_value(app, "mtcars", "gear", module_id = "module_1"))
+
+  navigate_teal_tab(app, "module_2")
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_identical(get_active_data_filters(app, "iris", module_id = "module_2"), "Species")
+  expect_identical(get_active_data_filters(app, "mtcars", module_id = "module_2"), c("drat", "gear"))
+  expect_identical(
+    get_active_selection_value(app, "iris", "Species", module_id = "module_2"),
+    c("setosa", "versicolor", "virginica")
+  )
+  expect_identical(
+    get_active_selection_value(app, "mtcars", "drat", is_numeric = TRUE, module_id = "module_2"),
+    c(3, 4)
+  )
+  expect_identical(
+    get_active_selection_value(app, "mtcars", "gear", module_id = "module_2"),
+    c("3", "4", "5")
+  )
+  expect_null(get_active_selection_value(app, "mtcars", "cyl", module_id = "module_2"))
+
+  set_active_selection_value(app, "iris", "Species", "setosa", module_id = "module_2")
+  navigate_teal_tab(app, "module_1")
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_identical(
+    get_active_selection_value(app, "iris", "Species", module_id = "module_1"),
+    "setosa"
+  )
+  app$stop()
+
+  # App with global filters
+  app <- get_test_app_object(
+    data = simple_teal_data,
+    modules = modules(
+      example_module(label = "Module_1"),
+      example_module(label = "Module_2")
+    ),
+    filter = teal_slices(
+      teal_slice(id = "iris_species", dataname = "iris", varname = "Species", multiple = TRUE),
+      teal_slice(id = "mtcars_cyl", dataname = "mtcars", varname = "cyl", selected = c(4, 6)),
+      teal_slice(id = "mtcars_drat", dataname = "mtcars", varname = "drat", selected = c(3, 4)),
+      teal_slice(id = "mtcars_gear", dataname = "mtcars", varname = "gear")
+    )
+  )
+
+  app$wait_for_idle(timeout = default_idle_timeout)
+
+  expect_identical(get_active_data_filters(app, "iris"), "Species")
+  expect_identical(get_active_data_filters(app, "mtcars"), c("cyl", "drat", "gear"))
+  expect_identical(
+    get_active_selection_value(app, "iris", "Species"),
+    c("setosa", "versicolor", "virginica")
+  )
+  expect_identical(
+    get_active_selection_value(app, "mtcars", "cyl"),
+    c("4", "6")
+  )
+  expect_identical(
+    get_active_selection_value(app, "mtcars", "drat", is_numeric = TRUE),
+    c(3, 4)
+  )
+  expect_identical(
+    get_active_selection_value(app, "mtcars", "gear"),
+    c("3", "4", "5")
+  )
+  app$stop()
+})
+
+test_that("module with a reporter creates the reporter tab", {
```

Comment:
two apps in a single test_that. Please split into separate test_that and adjust the description to the expectation

Reply:
The reason for having two test apps here was to test if the reporter appears only in a teal app with the reporter and it does not appear in a teal app without the reporter. I'll change the description so this expectation is clear.